### PR TITLE
Story AN.10: Improve Universe Table Editing UX and Data Handling

### LINKED ISSUE #399: Story AN.10: Wire Up Expired Filter - TDD GREEN Phase
## Story

**As a** user
**I want** to filter universe table to show/hide expired symbols
**So that** I can manage active vs expired symbols separately

## Context

**Current System:**
- Universe table displays data from SmartNgRX
- Previous stories have implemented base functionality
- Unit tests written in Story AN.9 (currently disabled)

**Implementation Approach:**
- Re-enable unit tests from AN.9
- Implement functionality to make tests pass
- Make all tests pass (GREEN phase)

### Functional Requirements
- [ ] All unit tests from AN.9 re-enabled and passing
- [ ] Tests cover all expected behaviors
- [ ] Implementation complete and working
- [ ] All edge cases handled properly

### Technical Requirements
- [ ] Tests follow existing testing patterns
- [ ] Mock dependencies properly configured
- [ ] Test coverage includes edge cases
- [ ] Implementation follows project patterns

### Step 1: Re-enable Tests
Remove `x` prefix or `.skip` from tests written in AN.9.

### Step 2: Implement Functionality
Implement the required functionality following TDD approach.

### Step 3: Verify All Tests Pass
```bash
pnpm test:dms-material
```

## Related Stories
- **Previous**: Story AN.9
- **Next**: Story AN.11
- **Epic**: Epic AN
- **Pattern Reference**: Story AM.10 (Similar TDD pattern)

### PR DESCRIPTION
**Pull request by DaveMBush**
## Summary

This PR implements improvements to the Universe table editing experience and fixes several data handling issues:

- **Removed spinners from numeric inputs** - Changed input type from "number" to "text" with `inputmode="decimal"` for cleaner UI
- **Fixed date editor timezone bug** - Dates now save correctly without day-off errors
- **Added keyboard shortcuts** - Enter to save, Escape to cancel in date editor  
- **Improved cell styling** - Editable fields now fit properly within table rows
- **Integrated SmartNgRX auto-save** - Cell edits automatically trigger server saves
- **Added data caching** - New UniverseService handles SmartNgRX state transitions
- **Updated column widths** - Ex-Date column: 175px, Distribution column: 120px
- **Enhanced risk group display** - Maps risk_group_id to human-readable names
- **Fixed Prisma validation** - Updated date format validation to accept ISO DateTime

### Verification Steps

1. Run the test suite: `pnpm test:dms-material`
2. Verify all 57 tests pass in filter-universes.function.spec.ts
3. Run validation commands:
   - `pnpm all`
   - `pnpm dupcheck`
   - `pnpm format`
4. Test in browser:
   - Navigate to Universe table
   - Edit distribution values - verify no spinner appears
   - Edit ex-date - verify date saves correctly (e.g., 02/19/2026 saves as 02/19/2026, not 02/18/2026)
   - Press Enter to save, Escape to cancel
   - Verify changes persist after page refresh

Closes #399

## Summary by CodeRabbit

* **New Features**
  * Keyboard shortcuts for date editing (Esc to cancel, Enter to save); background caching to keep the universe list stable during updates.

* **Bug Fixes**
  * Improved numeric input/parsing for amounts and yields; clearer date validation messages; corrected dividend date interpretation so displayed dates are accurate.

* **Improvements**
  * Risk group names now shown in the universe view with adjusted column widths; editable-date UI refinements (placeholder, hover) and decimal-capable inputs; expired-filter tests re-enabled.

* **Chores**
  * Updated yahoo‑finance2 dependency.

### CODE REVIEW

File: apps/dms-material/src/app/shared/components/editable-date-cell/editable-date-cell.component.ts [hunk 1]
```diff
@@ -32,13 +32,40 @@ import { MatInputModule } from '@angular/material/input';
...
+    } else {
+      this.internalValue = val;
+    }
+  }
+
+  get value(): Date | null {
+    return this.internalValue;
+  }
+
+  private internalValue: Date | null = null;
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/dms-material/src/app/shared/components/editable-date-cell/editable-date-cell.component.ts]: Looks like we are using signals now and this doesn't apply

File: apps/dms-material/src/app/account-panel/div-dep-modal/div-dep-modal.component.html [hunk 2]
```diff
@@ -28,7 +28,13 @@ <h2 mat-dialog-title>{{ title }}</h2>
...
       <mat-form-field appearance="outline" class="flex-1">
         <mat-label>Amount</mat-label>
-        <input matInput type="number" formControlName="amount" step="0.01" />
+        <input
+          matInput
+          type="text"
+          inputmode="decimal"
+          formControlName="amount"
+          step="0.01"
+        />
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/dms-material/src/app/account-panel/div-dep-modal/div-dep-modal.component.html]: fixed

File: apps/dms-material/src/app/global/global-universe/global-universe.columns.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { ColumnDef } from '../../shared/components/base-table/column-def.interfa
 export const UNIVERSE_COLUMNS: ColumnDef[] = [
   { field: 'symbol', header: 'Symbol', sortable: true, width: '80px' },
   {
-    field: 'risk_group_id',
+    field: 'risk_group',
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/dms-material/src/app/global/global-universe/global-universe.columns.ts]: fixed
